### PR TITLE
Always checkout by SHA

### LINKED ISSUE #69: Woodpecker keeps using files from older commit

### PR DESCRIPTION
**Pull request by qwerty287**
... if it's available

@xoxys Can you test this?

Trying to close #69

### CODE REVIEW

**Comment by xoxys** [comment]
Sure will try to test it later. Have do handle some real world tasks now 🙈 thanks for looking into it!

File: plugin.go [hunk 1]
```diff
@@ -69,11 +69,10 @@ func (p Plugin) Exec() error {
 	// fetch ref in any case
 	cmds = append(cmds, fetch(p.Build.Ref, p.Config.Tags, p.Config.Depth, p.Config.filter))
 
-	switch {
-	case isPullRequest(p.Build.Event) || isTag(p.Build.Event, p.Build.Ref) || p.Build.Commit == "":
+	if p.Build.Commit == "" {
```
Comment (6543) [inline plugin.go:72]: Can we also add a warning to stderr that the fallback was used

**Comment by xoxys** [comment]
What's the best way to test it? Do I need to build and push the image manually?

**Review by 6543 — APPROVED** [review]

**Comment by xoxys** [comment]
Ok, tested with a manual build, works for me. Tested with multiple commits adding/removing a line to a file, change get reflected immediately on CI.

**Pull request merged**